lib/super_digit.rb

```diff
@@ -1,15 +1,34 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)? - We made the recursive call 6 times to get to the answer for 9875. When the sum of the digits (sum) is less than 10, we will call the number of digits times (n). When 100 > sum > 10, we make n + 2 calls. When 1000 > sum > 100, we make n + 3 calls. Because of recursion, we need to run n times again to add up the digits. So it's O(2n) + something like O(log n) maybe??? and thus equals to O(n)?
+# Space Complexity - O(n) - call stack memory
+# I got this solution when refactoring my previous very chaotic solution. I don't feel like it's dynamic programming, but it seems clean. Basically, I had super_digit(9 / 10) = 0 here. I did try to make a variable sum to save to sum of the current digits, but didn't get it to work. Somehow it kept getting overwritten by n.
+# super_digit(9875)
+# 9875 % 10 + super_digit(9875 / 10) = 5 + 7 + 8 + 9 = 29
+#              = 987 % 10 + super_digit(987 / 10) = 7 + 8 + 9
+#                            = 98 % 10 + super_digit(98 / 10) = 8 + 9
+#                                         = 9 % 10 + super_digit(9 / 10) = 9
+# n = 29 because n > 9
+# 29 % 10 + super_digit(29 / 10) = 2 + 9
+#           = 9 % 10 super_digit(9 / 10) = 9
+
 def super_digit(n)
```

Comment:
This works but has O(n^2) time/space complexity because you have a sum of digits O(n) inside up to a number of recursive calls which is proportional to n, so O(n) recursive calls.

Reply:
I see it now. So every time we call super_digit(n) it will redo the sum from the base case to the current n, which makes its O(n^2) overall.